service/controller/clusterapi/v22/resource/clusterstatus/resource.go

```diff
@@ -16,13 +16,17 @@ type Config struct {
 	CMAClient clientset.Interface
 	G8sClient versioned.Interface
 	Logger    micrologger.Logger
+
+	Provider string
 }
 
 type Resource struct {
 	accessor  Accessor
 	cmaClient clientset.Interface
 	g8sClient versioned.Interface
 	logger    micrologger.Logger
+
+	provider string
```

Comment:
Can it be enum type? so don't have it as a string?

Reply:
It's consistent with other resources. If agreed we can refactor all in separate PR.

Reply:
I think it's better to have it in other PRs.

Reply:
We don't typically have enums for these flags and it is set via a configmap which is templated by draughtsman.

https://github.com/giantswarm/cluster-operator/blob/14bd41618c35eb3fe24b34e269a8d167b752bc61/helm/cluster-operator-chart/templates/configmap.yaml#L42

What is the benefit of using an enum here?

Reply:
if it's coming by configmap resource, I don't think it's really necessary to have it as enum.